lib/ContextModuleFactory.js

```diff
@@ -276,7 +276,26 @@ module.exports = class ContextModuleFactory extends ModuleFactory {
 		} = options;
 		if (!regExp || !resource) return callback(null, []);
 
-		const addDirectory = (directory, callback) => {
+		const addDirectoryChecked = (directory, visited, callback) => {
+			fs.realpath(directory, (err, realPath) => {
+				if (err) return callback(err);
+				if (visited.has(realPath)) return callback(null, []);
+				let nestedVisited;
+				addDirectory(
+					directory,
+					(dir, callback) => {
+						if (nestedVisited === undefined) {
```

Comment:
wouldn't `nestedVisited ` always be `undefined` here?.

Reply:
@sokra  it looks like it resolves  this case right, otherwise it'll just mark C as visited at the very first 
 ```
A
|
B
| \
N  C
|
C 
```